File: test_export.js
async function testExport() {
    // 1. Login
    const loginRes = await fetch("http://localhost:8080/api/v1/auth/login", {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify({ email: "minula@gmail.com", password: "admin123" })
    });
    
    if (!loginRes.ok) {
        console.error("Login failed:", loginRes.status, await loginRes.text());
        return;
    }
    
    const data = await loginRes.json();
    const token = data.data.token;
    console.log("Got token length:", token.length);
    
    // 2. Export
    const exportRes = await fetch("http://localhost:8080/api/v1/reports/export/sales", {
        headers: { "Authorization": `Bearer ${token}` }
    });
    
    console.log("Export status:", exportRes.status);
    console.log("Headers:", [...exportRes.headers.entries()]);
    
    const text = await exportRes.text();
    console.log("Content length:", text.length);
    console.log("Preview:", text.substring(0, 200));
}

testExport();

File: test_export.py
import requests
import json

def test_export():
    # 1. Login to get token
    login_data = {"email": "minula@gmail.com", "password": "admin123"}
    r = requests.post("http://localhost:8080/api/v1/auth/login", json=login_data)
    if r.status_code != 200:
        print(f"Login failed: {r.status_code} {r.text}")
        return
        
    token = r.json().get("data", {}).get("token")
    if not token:
        print("No token in response")
        return
        
    print(f"Got token! Length: {len(token)}")
    
    # 2. Call export endpoint
    headers = {"Authorization": f"Bearer {token}"}
    r2 = requests.get("http://localhost:8080/api/v1/reports/export/sales", headers=headers)
    
    print(f"Export HTTP Status: {r2.status_code}")
    print(f"Export Headers: {r2.headers}")
    print(f"Content Length: {len(r2.content)}")
    print(f"Content preview: {r2.text[:200]}")

if __name__ == "__main__":
    test_export()
